src/mlpack/tests/main_tests/krann_test.cpp

```diff
@@ -0,0 +1,479 @@
+
+  // Reset passed parameters.
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["query"].wasPassed = false;
+
+  // Input saved model, pass the same query and keep k unchanged.
+  SetInputParam("input_model", output_model);
+  SetInputParam("query", queryData);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Check that initial output matrices and the output matrices using
+  // saved model are equal.
+  CheckMatrices(neighbors, CLI::GetParam<arma::Mat<size_t>>("neighbors"));
+  CheckMatrices(distances, CLI::GetParam<arma::mat>("distances"));
+}
+
+/**
+  * Ensure that different leaf sizes give different results.
+ */
+BOOST_AUTO_TEST_CASE(KRANNDifferentLeafSizes)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of top tau %ile reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 5);
+  SetInputParam("leaf_size", (int) 1);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  RANNModel* output_model;
+  output_model = std::move(CLI::GetParam<RANNModel*>("output_model"));
+
+  // Reset passed parameters.
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+
+  // Input saved model, pass the same query and keep k unchanged.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+  SetInputParam("leaf_size", (int) 10);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Check that initial output matrices and the output matrices using
+  // saved model are equal.
+  BOOST_CHECK_EQUAL(output_model->LeafSize(), (int) 1);
+  BOOST_CHECK_EQUAL(CLI::GetParam<RANNModel*>("output_model")->LeafSize(),
+      (int) 10);
+  delete output_model;
+}
+
+/**
+ * Ensure that different tau give different results.
+ */
+BOOST_AUTO_TEST_CASE(KRANNDifferentTau)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Random input, some k <= number of top tau %ile reference points.
+  SetInputParam("reference", referenceData);
+  SetInputParam("k", (int) 5);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+  
+  RANNModel* output_model;
+  output_model = std::move(CLI::GetParam<RANNModel*>("output_model"));
+  
+  // Reset the passed  parameters.
+  CLI::GetSingleton().Parameters()["reference"].wasPassed = false;
+
+  // Changing value of tau and keeping everything else unchanged.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 5);
+  SetInputParam("tau", (double) 10);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Check that initial output matrices and the output matrices using
+  // saved model are equal
+  BOOST_CHECK_EQUAL(output_model->Tau(), (double) 5);
+  BOOST_CHECK_EQUAL(CLI::GetParam<RANNModel*>("output_model")->Tau(),
+        (double) 10);
```

Comment:
```suggestion
      (double) 10);
```
(same thing as the next comment below)